pkg/resources/cloudcontroller/vsphere.go

```diff
@@ -76,7 +76,6 @@ func vsphereDeploymentCreator(data *resources.TemplateData) reconciling.NamedDep
 				return nil, err
 			}
 			dep.Spec.Template.Spec.AutomountServiceAccountToken = pointer.BoolPtr(false)
-			dep.Spec.Template.Spec.HostNetwork = true
```

Comment:
Why did we remove this? For reference, the upstream manifest has `hostNetwork: true`: 

https://github.com/kubernetes/cloud-provider-vsphere/blob/951214ce27ed7dc439a812c4ab27c16947f18544/releases/v1.21/vsphere-cloud-controller-manager.yaml#L249

Reply:
I removed it because the CCM works with the user cluster but is deployed in the seed one, and having the CCM in the host network in the seed cluster doesn't make any sense. We do the same thing for [openstack](https://github.com/kubernetes/cloud-provider-openstack/blob/295f57c9f0af38d50b901db2414de2114d18208f/charts/openstack-cloud-controller-manager/templates/daemonset.yaml#L84). Besides if we set the pod in `hostNetwork`, and we configure the DNS as it is required to be, the entire dns configuration of the node messes up (this is the reason behind the periodic vsphere test failure).